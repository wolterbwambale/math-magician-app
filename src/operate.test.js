import operate from './logic/operate';

describe('operate', () => {
  it('should perform addition correctly', () => {
    const result = operate('10', '5', '+');
    expect(result).toMatchSnapshot();
  });

  it('should perform subtraction correctly', () => {
    const result = operate('10', '5', '-');
    expect(result).toMatchSnapshot();
  });

  it('should perform multiplication correctly', () => {
    const result = operate('10', '5', 'x');
    expect(result).toMatchSnapshot();
  });

  it('should perform division correctly', () => {
    const result = operate('10', '5', 'รท');
    expect(result).toMatchSnapshot();
  });

  it('should handle division by zero', () => {
    const result = operate('10', '0', 'รท');
    expect(result).toMatchSnapshot();
  });

  it('should perform modulo correctly', () => {
    const result = operate('10', '5', '%');
    expect(result).toMatchSnapshot();
  });

  it('should handle modulo with zero divisor', () => {
    const result = operate('10', '0', '%');
    expect(result).toMatchSnapshot();
  });
});
