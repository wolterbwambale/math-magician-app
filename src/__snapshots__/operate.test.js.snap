// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`operate should handle division by zero 1`] = `"Can't divide by 0."`;

exports[`operate should handle modulo with zero divisor 1`] = `"Can't find modulo as can't divide by 0."`;

exports[`operate should perform addition correctly 1`] = `"15"`;

exports[`operate should perform division correctly 1`] = `"2"`;

exports[`operate should perform modulo correctly 1`] = `"0"`;

exports[`operate should perform multiplication correctly 1`] = `"50"`;

exports[`operate should perform subtraction correctly 1`] = `"5"`;
